File: Machine Learning A-Z Template Folder/Part 1 - Data Preprocessing/data_preprocessing_template.R
data_set = read.csv('/Users/rmaffesoli/Documents/personal/ML_A_Z/Machine Learning A-Z Template Folder/Part 1 - Data Preprocessing/Data.csv')
# data_set = data_set[, 2:3]

data_set$Age = ifelse(is.na(data_set$Age),
                      ave(data_set$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                      data_set$Age)
data_set$Salary = ifelse(is.na(data_set$Salary),
                         ave(data_set$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                         data_set$Salary)

data_set$Country = factor(data_set$Country,
                          levels= c('France', 'Spain', 'Germany'),
                          labels = c(1, 2, 3))

data_set$Purchased = factor(data_set$Purchased,
                            levels = c('No', 'Yes'),
                            labels= c(0,1))

# install.packages('caTools', repos = "http://cran.us.r-project.org")
library(caTools)
set.seed(123)
split = sample.split(data_set$Purchased, SplitRatio = 0.8)
training_set = subset(data_set, split == TRUE)
test_set = subset(data_set, split == FALSE)

# training_set[ ,2:3] = scale(training_set[ ,2:3])
# test_set[ ,2:3] = scale(test_set[ ,2:3])

print(training_set)
print(test_set)

File: Machine Learning A-Z Template Folder/Part 1 - Data Preprocessing/data_preprocessing_template.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import Imputer, LabelEncoder, OneHotEncoder, StandardScaler
from sklearn.model_selection import train_test_split


data_set = pd.read_csv('/Users/rmaffesoli/Documents/personal/ML_A_Z/Machine Learning A-Z Template Folder/Part 1 - Data Preprocessing/Data.csv')
X = data_set.iloc[:, :-1].values
y = data_set.iloc[:, -1].values

imputer = Imputer(missing_values='NaN', strategy="mean", axis=0)
imputer = imputer.fit(X[:, 1:3])
X[:, 1:3] = imputer.transform(X[:, 1:3])

label_encoder_X = LabelEncoder()
X[:, 0] = label_encoder_X.fit_transform(X[:, 0])

one_hot_encoder = OneHotEncoder(categorical_features=[0])
X = one_hot_encoder.fit_transform(X).toarray()

label_encoder_y = LabelEncoder()
y = label_encoder_y.fit_transform(y)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)


print(X_train, X_test)
# print(y_train, y_test)
